### PR TITLE
Raciones list pod

### PR DESCRIPTION
**Pull request by Pdominguez15**

### CODE REVIEW

File: admin/src/pods/raciones-list/raciones-list.container.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+//VM
+import { Racion, createEmptyRacion } from './racion.vm';
+
+//Mapper
+import { mapRacionesListFromApiToVm, mapRacionFromVmToApi } from './raciones-list.mapper';
+
+//Componente
+import { RacionesListComponent } from './raciones-list.component';
+
+//Metodo
```
Comment (brauliodiez) [inline admin/src/pods/raciones-list/raciones-list.container.tsx:15]: Please set the comment in english

File: admin/src/pods/raciones-list/raciones-list.container.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  const onLoadRaciones = async () => {
+    api
+      .getRacionesList()
+      .then((result) => {
+        setRaciones(mapRacionesListFromApiToVm(result));
+      })
+      .catch((error) => {
+        // Snackbar error
+        alert('Error to load raciones list');
```
Comment (brauliodiez) [inline admin/src/pods/raciones-list/raciones-list.container.tsx:34]: Error loading raciones list

We should add a better error handling, I would open this as a separate issue

File: admin/src/pods/raciones-list/raciones-list.container.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    setRaciones(reorder(raciones, startIndex, endIndex));
+
+  const handleSave = (value: string, id: number) => {
+    const newRacionAPI = mapRacionFromVmToApi({ id, value });
+
+    api
+      .saveRacion(newRacionAPI)
+      .then((result) => {
+        // Snackbar
+        alert('Saved racion');
```
Comment (brauliodiez) [inline admin/src/pods/raciones-list/raciones-list.container.tsx:48]: Let's create a new issue to handle notifications

File: admin/src/pods/raciones-list/raciones-list.container.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        setEditRacionId(false);
+        const index = raciones.findIndex((racion) => racion.id === id);
+        if (index !== -1) {
+          const newArray = [...raciones];
+          newArray[index].value = value;
+          console.log(newArray);
```
Comment (brauliodiez) [inline admin/src/pods/raciones-list/raciones-list.container.tsx:54]: Remove console.log

File: admin/src/pods/raciones-list/raciones-list.container.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+          setRaciones(newArray);
+        } else {
+          setRaciones([...raciones, { id, value }]);
+        }
```
Comment (brauliodiez) [inline admin/src/pods/raciones-list/raciones-list.container.tsx:58]: Rather use immer here

File: admin/src/pods/raciones-list/raciones-list.mapper.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+export const mapRacionFromApiToVm = (racion: apiModel.Racion): viewModel.Racion => ({
+  id: racion.id,
+  value: racion.name,
+});
+
+export const mapRacionFromVmToApi = (racion: viewModel.Racion): apiModel.Racion => ({
+  id: racion.id,
+  name: racion.value,
+});
```
Comment (brauliodiez) [inline admin/src/pods/raciones-list/raciones-list.mapper.ts:19]: Add specs (tests) to all this mappers, check null, undefined ... be defensive

File: admin/src/pods/raciones-list/raciones-list.container.tsx (hunk 1 above)
Comment (Pdominguez15) [reply admin/src/pods/raciones-list/raciones-list.container.tsx:15]: Fixed

File: admin/src/pods/raciones-list/raciones-list.container.tsx (hunk 2 above)
Comment (Pdominguez15) [reply admin/src/pods/raciones-list/raciones-list.container.tsx:34]: Ok, I wait for that issue then

File: admin/src/pods/raciones-list/raciones-list.container.tsx (hunk 3 above)
Comment (Pdominguez15) [reply admin/src/pods/raciones-list/raciones-list.container.tsx:48]: Assign it to me if you want

File: admin/src/pods/raciones-list/raciones-list.container.tsx (hunk 4 above)
Comment (Pdominguez15) [reply admin/src/pods/raciones-list/raciones-list.container.tsx:54]: Removed

File: admin/src/pods/raciones-list/raciones-list.container.tsx (hunk 5 above)
Comment (Pdominguez15) [reply admin/src/pods/raciones-list/raciones-list.container.tsx:58]: It´s a simple mock, when we change for the real api i will delete this code

File: admin/src/pods/raciones-list/raciones-list.mapper.ts (hunk 6 above)
Comment (Pdominguez15) [reply admin/src/pods/raciones-list/raciones-list.mapper.ts]: Resolved
